src/bits/mod.rs

```diff
@@ -1,17 +1,14 @@
-use crate::secret_sharing::BooleanShare;
-
+use crate::secret_sharing::SharedValue;
 use generic_array::{ArrayLength, GenericArray};
-
 use std::ops::{BitAnd, BitAndAssign, BitOr, BitOrAssign, BitXor, BitXorAssign, Index, Not};
 
-mod bit_array;
-
-pub use bit_array::{BitArray40, BitArray8};
+mod shared_value_array;
+pub use shared_value_array::{BitArray40, BitArray8};
 
 /// Trait for data types storing arbitrary number of bits.
 // TODO: Implement `Message`
```

Comment:
Can we not implement serializable generically?

Reply:
@akoshelev can we? I can make the change together in this diff if we could.

Reply:
I would do this in a later PR, because it is distinct from this.